### PR TITLE
use disallow-any-generics mypy flag

### PR DESCRIPTION
**Pull request by vtomole**
Part of https://github.com/SupertechLabs/general-superstaq/issues/68

### CODE REVIEW

File: general_superstaq/check/build_docs.py [hunk 1]
```diff
@@ -27,7 +27,9 @@ def run(*args: str, sphinx_paths: Optional[List[str]] = None) -> int:
         if sphinx_paths:
             for path in sphinx_paths:
                 subprocess.run(
-                    f"sphinx-apidoc -f -o source {path} {path}/*_test.py", shell=True, cwd=docs_dir
+                    f"sphinx-apidoc -f -o source {path} {path}/*_test.py",
+                    shell=True,
+                    cwd=docs_dir,
```
Comment (richrines1) [inline general_superstaq/check/build_docs.py:32]: why all these formatting changes? the previous versions seem to pass checks

Comment (vtomole) [reply general_superstaq/check/build_docs.py:32]: Reverted.

Comment (vtomole) [reply general_superstaq/check/build_docs.py]: Reverted.

File: general_superstaq/superstaq_client.py [hunk 2]
```diff
@@ -230,28 +232,32 @@ def submit_qubo(self, qubo: qv.QUBO, target: str, repetitions: int = 1000) -> di
...
         """Makes a POST request to SuperstaQ API to find a optimal TSP tour."""
         return self.post_request("/tsp", json_dict)
 
-    def warehouse(self, json_dict: dict) -> dict:
+    def warehouse(self, json_dict: Dict[str, Union[int, List[str], str]]) -> WareHouseJson:
         """Makes a POST request to SuperstaQ API to find optimal warehouse assignment."""
         return self.post_request("/warehouse", json_dict)
 
-    def aqt_upload_configs(self, aqt_configs: Dict[str, str]) -> dict:
+    def aqt_upload_configs(self, aqt_configs: Dict[str, str]) -> Dict[str, str]:
```
Comment (richrines1) [inline general_superstaq/superstaq_client.py:256]: think this one should just be
```suggestion
    def aqt_upload_configs(self, aqt_configs: Dict[str, str]) -> str:
```
(the docstring in `UserConfig.aqt_upload_configs` is also wrong though)

Comment (vtomole) [reply general_superstaq/superstaq_client.py]: Updated the docstring

Comment (vtomole) [reply general_superstaq/superstaq_client.py]: Committed the suggestion.

File: general_superstaq/finance.py [hunk 3]
```diff
@@ -147,7 +149,7 @@ def find_max_pseudo_sharpe_ratio(
             .best_std_dev: The volatility of the optimal portfolio.
             .best_sharpe_ratio: The Sharpe ratio of the optimal portfolio.
         """
-        input_dict = {
+        input_dict: Dict[str, Union[List[str], float, str, Optional[int]]] = {
```
[the remark this answers is not in the document]
Comment (vtomole) [reply general_superstaq/finance.py]: Done.

File: general_superstaq/superstaq_client.py [hunk 4]
```diff
@@ -198,30 +198,32 @@ def request() -> requests.Response:
 
         return self._make_request(request).json()
 
-    def resource_estimate(self, json_dict: Dict[str, str]) -> dict:
+    def resource_estimate(self, json_dict: Dict[str, str]) -> Dict[str, List[Dict[str, int]]]:
         return self.post_request("/resource_estimate", json_dict)
 
-    def aqt_compile(self, json_dict: Dict[str, Union[int, str, List[str]]]) -> dict:
+    def aqt_compile(self, json_dict: Dict[str, Union[int, str, List[str]]]) -> Dict[str, str]:
```
[the remark this answers is not in the document]
Comment (vtomole) [reply general_superstaq/superstaq_client.py:204]: Will leave as-is

File: general_superstaq/logistics.py [hunk 5]
```diff
@@ -91,7 +90,11 @@ def tsp(self, locs: List[str], solver: str = "anneal") -> TSPOutput:
         return read_json_tsp(json_dict)
 
     def warehouse(
-        self, k: int, possible_warehouses: List[str], customers: List[str], solver: str = "anneal"
+        self,
+        k: int,
+        possible_warehouses: List[str],
+        customers: List[str],
+        solver: str = "anneal",
```
Comment (richrines1) [inline general_superstaq/logistics.py:97]: nit
```suggestion
        self, k: int, possible_warehouses: List[str], customers: List[str], solver: str = "anneal"
```

File: general_superstaq/finance_test.py [hunk 6]
```diff
@@ -72,7 +77,9 @@ def test_service_submit_qubo(mock_submit_qubo: mock.MagicMock) -> None:
         "qubo": [{"keys": ["0"], "value": 123}],
     },
 )
-def test_service_find_min_vol_portfolio(mock_find_min_vol_portfolio: mock.MagicMock) -> None:
+def test_service_find_min_vol_portfolio(
+    mock_find_min_vol_portfolio: mock.MagicMock,
+) -> None:
```
Comment (richrines1) [inline general_superstaq/finance_test.py:82]: nit
```suggestion
def test_service_find_min_vol_portfolio(mock_find_min_vol_portfolio: mock.MagicMock) -> None:
```

**Review by richrines1 — APPROVED** [review]
lg w/ two more nits

File: general_superstaq/logistics.py (hunk 5 above)
Comment (vtomole) [reply general_superstaq/logistics.py]: Done.

File: general_superstaq/finance_test.py (hunk 6 above)
Comment (vtomole) [reply general_superstaq/finance_test.py]: Done.

**Comment by vtomole** [comment]
Won't merge until i've confirmed that this causes no problems with [css]https://github.com/SupertechLabs/cirq-superstaq/pull/338) and [qss](https://github.com/SupertechLabs/qiskit-superstaq/pull/234)

**Pull request merged**